projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/RunJob.tsx

```diff
@@ -71,41 +71,139 @@ export default class RunJobDialog extends Component<IJobPathProp> {
   };
 }
 
-export async function showRunJobDialog() {
+export async function showRunJobDialog(docManager?: IDocumentManager) {
   const result = await showDialog({
     title: 'Run Job',
-    body: (
-      <RunJobDialog
-        jobPath={jobData.get(VdkOption.PATH)!}
-      ></RunJobDialog>
-    ),
+    body: <RunJobDialog jobPath={jobData.get(VdkOption.PATH)!}></RunJobDialog>,
     buttons: [Dialog.okButton(), Dialog.cancelButton()]
   });
   if (result.button.accept) {
     let { message, status } = await jobRunRequest();
-        if (status) {
-          showDialog({
-            title: 'Run Job',
-            body: <div className='vdk-run-dialog-message-container'>
-            <p className='vdk-run-dialog-message'>The job was executed successfully!</p>
-          </div>,
-            buttons: [Dialog.okButton()]
-          });
-        }
-        else{
-          message = "ERROR : " + message;
-          const  errorMessage = new VdkErrorMessage(message);
-          showDialog({
-            title: 'Run Job',
-            body: <div  className="vdk-run-error-message ">
+    if (status) {
+      showDialog({
+        title: 'Run Job',
+        body: (
+          <div className="vdk-run-dialog-message-container">
+            <p className="vdk-run-dialog-message">
+              The job was executed successfully!
+            </p>
+          </div>
+        ),
+        buttons: [Dialog.okButton()]
+      });
+    } else {
+      message = 'ERROR : ' + message;
+      const errorMessage = new VdkErrorMessage(message);
+      if (
+        !docManager ||
```

Comment:
Nit: It might be better to export this condition to a variable with a proper name, so it's clear what it does. Decomposing the condition will also help.

```
const errorsHandled = docManager && (await handleErrorsProducedByNotebookCell(errorMessage, docManager))

if (!errorsHandled) {
  // do stuff
}
```